sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -315,4 +316,142 @@ def test_pinjoint_pi():
+    N, A, P, C = _generate_body()
+    PrismaticJoint('S', P, C, child_axis=A.y, child_joint_pos=A.x)
+
+    assert A.y.angle_between(N.x) == 0 #Axis are aligned
+    assert A.y.express(N) == N.x
+    assert A.dcm(N) == Matrix([[0, -1, 0], [1, 0, 0], [0, 0, 1]])
+    assert C.masscenter.vel(N) == v * N.x
+    assert C.masscenter.vel(N).express(A) == v * A.y
+    assert C.masscenter.pos_from(P.masscenter) == x*N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == x*N.x + N.y
+    assert A.ang_vel_in(N) == 0
+    assert N.ang_vel_in(A) == 0
+
+    #Similar to previous case but wrt parent body
+    N, A, P, C = _generate_body()
+    PrismaticJoint('S', P, C, parent_axis=N.y, parent_joint_pos=N.x)
+
+    assert N.y.angle_between(A.x) == 0 #Axis are aligned
+    assert N.y.express(A) ==  A.x
+    assert A.dcm(N) == Matrix([[0, 1, 0], [-1, 0, 0], [0, 0, 1]])
+    assert C.masscenter.vel(N) == v * N.y
+    assert C.masscenter.vel(N).express(A) == v * A.x
+    assert C.masscenter.pos_from(P.masscenter) == N.x + x*N.y
+    assert A.ang_vel_in(N) == 0
+    assert N.ang_vel_in(A) == 0
+
+    #Both joint pos id defined but different axes
+    N, A, P, C = _generate_body()
+    PrismaticJoint('S', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y)
+    assert N.x.angle_between(A.x + A.y) == 0 #Axis are aligned
+    assert (A.x + A.y).express(N) == sqrt(2)*N.x
+    assert A.dcm(N) == Matrix([[sqrt(2)/2, -sqrt(2)/2, 0], [sqrt(2)/2, sqrt(2)/2, 0], [0, 0, 1]])
+    assert C.masscenter.pos_from(P.masscenter) == (x + 1)*N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N) == (x - sqrt(2)/2 + 1)*N.x + sqrt(2)/2*N.y
+    assert C.masscenter.vel(N).express(A) == v * (A.x + A.y)/sqrt(2)
+    assert C.masscenter.vel(N) == v*N.x
+    assert A.ang_vel_in(N) == 0
+    assert N.ang_vel_in(A) == 0
+
+    N, A, P, C = _generate_body()
+    PrismaticJoint('S', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y-A.z)
+    assert N.x.angle_between(A.x + A.y - A.z) == 0 #Axis are aligned
+    assert (A.x + A.y - A.z).express(N) == sqrt(3)*N.x
+    assert A.dcm(N) == Matrix([[sqrt(3)/3, -sqrt(3)/3, sqrt(3)/3],
+                                [sqrt(3)/3, sqrt(3)/6 + S(1)/2, S(1)/2 - sqrt(3)/6],
+                                [-sqrt(3)/3, S(1)/2 - sqrt(3)/6, sqrt(3)/6 + S(1)/2]])
+    assert C.masscenter.pos_from(P.masscenter) == (x + 1)*N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N) == \
+        (x - sqrt(3)/3 + 1)*N.x + sqrt(3)/3*N.y - sqrt(3)/3*N.z
+    assert C.masscenter.vel(N) == v*N.x
+    assert C.masscenter.vel(N).express(A) == sqrt(3)*v/3*A.x + sqrt(3)*v/3*A.y - sqrt(3)*v/3*A.z
+    assert A.ang_vel_in(N) == 0
+    assert N.ang_vel_in(A) == 0
+
+    N, A, P, C = _generate_body()
+    m, n = symbols('m n')
+    PrismaticJoint('S', P, C, parent_joint_pos=m*N.x, child_joint_pos=n*A.x,
+                    child_axis=A.x+A.y-A.z, parent_axis=N.x-N.y+N.z)
```

Comment:
@moorepants My figure for this case

![vector](https://user-images.githubusercontent.com/45732670/125037336-a232bf00-e0b1-11eb-84e8-58201a56bc30.jpg)

The angle between the axes is theta(constant angle), so the child frame is rotated at an angle theta, wrt parent_frame where rotation_axis=cross(parent_axis, parent_frame.x)

Reply:
The figure I was asking for is one that shows the position vectors (as I did in the figure I shared above).

Reply:
![slider](https://user-images.githubusercontent.com/45732670/126030872-6a744e35-b0b2-4713-946e-59b5032cf2e9.jpg)

@moorepants I hope I got it right this time.

Reply:
Does this example imply that:

```
N.x = A.y
N.y = A.z
N.z = A.x
```